pinot-server/src/main/java/com/linkedin/pinot/server/starter/helix/HelixServerStarter.java

```diff
@@ -220,10 +223,38 @@ public void stop() {
     if (_helixServerConfig.getBoolean(CommonConstants.Server.CONFIG_OF_ENABLE_SHUTDOWN_DELAY, true)) {
       Uninterruptibles.sleepUninterruptibly(_maxQueryTimeMs, TimeUnit.MILLISECONDS);
     }
+    waitUntilNoIncomingQueries();
```

Comment:
From our observations and evidence from logs, broker sees instance config updates soon enough and the server will stop getting requests soon after shuttingdown is set to true. I am not sure this change will actually get what we want.